templates/layout.html

```diff
@@ -6,7 +6,8 @@
 </head>
 <body>
     {% if session.logged_in %}
-        <p>Hello {{ session.first_name }} {{ session.name }}. <a href="{{ url_for('auth.logout') }}">Log out</a></p>
+        <!--- <p>Hello {{ session.first_name }} {{ session.name }}. <a href="{{ url_for('auth.logout') }}">Log out</a></p> --->
+        {%  extends "home.html" %}
```

Comment:
This is probably not what you meant to do. Template inheritance in Jinja works as follow :

layout.html :

    <html>
    <head>
        <script src="bootstrap.js"></script>
        <link href="bootstrap.css" rel="stylesheet" />
        {% block additionalpageheader %}{% endblock %}
        <title>{% block pagetitle %}{% endblock %} | ICTM Teaching</title>
    </head>
    <body>
        <div id="menu">
             <!-- Some links  -->
        </div>
        <div id="content">
            {% block pagecontent %}{%endblock %}
        </div>
    </body>
    </html>

home.html:

    {% extends "layout.html" %}
    {% block pagetitle %}Welcome{% endblock %}
    {% block pagecontent %}Hello world ! Please navigate this app trough the menu.{% endblock %}

You should not repeat the basic layout code in each template but only adds what is specific to the view. Block `additionalpageheader` can be defined in the child templates to add specific JS or CSS files only required for a specific view. The same thing can be done for the footer.